RealmSwift/SwiftUI.swift

```diff
@@ -0,0 +1,395 @@
+        return value.observe { [binder] _ in
+            block(binder)
+        }
+    }
+}
+
+@dynamicMemberLookup
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+public final class AnyBound<T>: Bound, ObservableObject {
+    public init(_ value: T) {
+        self.value = value
+    }
+
+    var value: T
+    var realm: Realm?
+    var _observe: ((inout RealmBind<T>, _ block: @escaping (RealmBind<T>) -> Void) -> NotificationToken)? = nil
+
+    init(_ value: T, realm: Realm?) where T: Object {
+        self.value = value
+        self._observe = BoundObject(value).observe
+        self.realm = realm
+    }
+    init<Element>(_ value: T, realm: Realm?) where T == Results<Element> {
+        self.value = value
+        self._observe = BoundResults(value).observe
+        self.realm = realm
+    }
+
+    func observe(binder: inout RealmBind<T>, _ block: @escaping (RealmBind<T>) -> Void) -> NotificationToken {
+        _observe!(&binder) { [block] binder in
+            block(binder)
+        }
+    }
+
+    public subscript<V>(dynamicMember member: ReferenceWritableKeyPath<T, V>) -> Binding<V> {
+        get {
+            Binding(get: {
+                self.value[keyPath: member]
+            }, set: { newValue in
+                try! self.realm!.write {
+                    self.value[keyPath: member] = newValue
+                }
+            })
+        }
+    }
+
+    // if property is a list, we want to freeze it before fetching it
+    public subscript<V>(dynamicMember member: ReferenceWritableKeyPath<T, V>) -> Binding<V> where V: ListBase {
+        get {
+            Binding(get: {
+                self.value[keyPath: member].freeze()
+            }, set: {
+                self.value[keyPath: member] = $0
+            })
+        }
+    }
+}
+
+// MARK: Bindings
+
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+public extension Binding where Value: Object {
+    subscript<V>(dynamicMember member: ReferenceWritableKeyPath<Value, V>) -> Binding<V> {
+        get {
+            Binding<V>(get: {
+                self.wrappedValue[keyPath: member]
+            }, set: { newValue in
+                try! wrappedValue.realm!.write {
+                    self.wrappedValue[keyPath: member] = newValue
+                }
+            })
+        }
+    }
+
+    subscript<V>(dynamicMember member: ReferenceWritableKeyPath<Value, V>) -> Binding<V> where V: ListBase {
+        get {
+            Binding<V>(get: {
+                self.wrappedValue[keyPath: member].freeze()
+            }, set: {
+                self.wrappedValue[keyPath: member] = $0
+            })
+        }
+    }
+}
+
+
+// Allows conformance with SwiftUI's `ForEach`
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+extension Binding: Identifiable where Value: RealmCollectionValue, Value: Identifiable {
```

Comment:
Declaring conditional conformances on other people's types is a no-no. There can only be one conformance for a type to a protocol, so this will break if another library (or the app) tries to do the same thing.

Reply:
For `Identifiable`, I take your point, as that may be common (though it will make getting the `\.id` key path more awkward for the end user).

But for the List binding, can we simply say that's a cost of using Realm with SwiftUI? A user wanting to conform `Binding` to `RandomAccessCollection`, `Sequence`, `Collection`, `BidirectionalCollection` seems like it would be highly unlikely?

I have to think about this. There might be an alternative path here (though I fear this was a fundamental part of my implementation).

Reply:
Suppose we may not need to extend Binding at all...